serverv2/mempool/mempool.go

```diff
@@ -0,0 +1,25 @@
+package mempool
+
+import (
+	"context"
+
+	tx "cosmossdk.io/core/transaction"
+)
+
+// Mempool defines the required methods of an application's mempool.
+type Mempool[T tx.Tx] interface {
+	// Insert attempts to insert a Tx into the app-side mempool returning
+	// an error upon failure. Insert will validate the transaction using the txValidator
+	Insert(ctx context.Context, txs T) map[[32]byte]error
+
+	// GetTxs returns a list of transactions to add in a block
+	// size specifies the size of the block left for transactions
+	GetTxs(ctx context.Context, size uint32) (ts any, err error)
```

Comment:
There's an issue open (https://github.com/cosmos/cosmos-sdk/issues/18547) about the SDK calculating a tx's size differently than the consensus engine.
Maybe we should allow a `TxSizeFn` to be passed to the mempool?

Reply:
As an argument to `GetTxs`, yeah.